### ISSUE TITLE
Unable to install new version of DropletUtils

### ISSUE THREAD

**Issue by liron27** [MarioniLab/DropletUtils#52]
Hi,
I have tried to install the new version of the package  using:
devtools::install_github("MarioniLab/DropletUtils")

and got the following error:

ERROR: compilation failed for package ‘DropletUtils’
* removing ‘/tmp/Rtmp4VjwFR/Rinst19fd655306c7/DropletUtils’

Thanks

**Comment by LTLA** [comment]
Try installing using the recommended instructions on the Bioconductor landing page.

**Comment by liron27** [comment]
I did but it doesn't include the new function  removeAmbience

**Comment by LTLA** [comment]
You can either wait a few hours for the new builds to propagate on Bioconductor. Or you can give me much more detail about the entire error log, I can't figure out what's going on from just the last line.

**Comment by liron27** [comment]
I just copied the removeAmbience function from github to my local directory and then I used the following code:

raw.count <- Read10X(data.dir = "path.to.raw.count")
ambient <- estimateAmbience(raw.count)
x<-removeAmbience(raw.count,ambient, colnames(raw.count))

I am not sure if this is the way to go, but it is still running (for almost 20 minutes now...)

**Comment by LTLA** [comment]
Stop, stop stop. Hold up. You really need to give it clusters. Did you read the documentation?

Install the package properly rather than doing whatever you're doing.

**Comment by liron27** [comment]
I read, but have few issues:

1. Doing the clustering on the original raw data (without cleaning ambient
rna) is computationally heavy - my original raw data has over 6 millions
barcodes ...

2. Should I run the entire pipeline of normalization, PCA, clustering on
the raw data and then bring it back to “removeAmbience”?

Thanks!

**Comment by LTLA** [comment]
I doubt you have 6 million _cells_ generated from a single run of a 10X machine. The maximum number of barcodes is around 700k, at least for versions 2 and 3 of the 10X chemistry.

**Comment by liron27** [comment]
Yep, I suspected it’s an issue... I am using the “Read10X” function with a
path to the directory that contains the  barcodes, features and matrix.

**Comment by LTLA** [comment]
The book build crashed so you won't be able to see the latest chapter of the book, but you can just read it from the Rmarkdown starting at:

https://github.com/Bioconductor/OrchestratingSingleCellAnalysis/blob/092880538f4c475fd397c6ab390a69027c545cdc/droplet-processing.Rmd#L137

Or you can just wait for Friday, hopefully the book will have built by then.

**Comment by liron27** [comment]
Thank you!

**Comment by liron27** [comment]
Regarding the 6 million barcode issues from 10x - 10x increased the barcode
# in the whitelist in V3 version to roughly ~3.5 Million BCs. I believe the
barcodes.tsv.gz contains barcode sequences and their reverse complement
sequences, so making the total # ~6.8M.

**Comment by LTLA** [comment]
I would be shocked if all 6 million barcodes had counts. Even looking at the existing output that I get from CellRanger 3, I only get non-zero counts for ~600k barcodes.

**Comment by liron27** [comment]
This is from 10x genomics


I’m using the makecount command to generate both the unfiltered and
filtered matrix from the fastq file.

**Comment by LTLA** [comment]
Did you resolve your issue?

**Comment by liron27** [comment]
No
